src/encoded/types/file.py

```diff
@@ -1231,140 +1231,98 @@ def download(context, request):
-        'downloads'         : '2'
-    }
 
-    ga_cid = request.cookies.get("clientIdentifier")
-    if not ga_cid: # Fallback, potentially can stop working as GA is updated
-        ga_cid = request.cookies.get("_ga")
+        # client id (`cid`) or user id (`uid`) is required. uid shall be user uuid.
+        # client id might be gotten from Google Analytics cookie, but not stable to use and wont work on programmatic requests...
+        if user_uuid:
+            ga_payload['uid'] = user_uuid
         if ga_cid:
-            ga_cid = ".".join(ga_cid.split(".")[2:])
-
-    ga_tid = ga_host_map.get(request.host, ga_host_map["default"])
-    # We're sending 2 things here, an Event and a Transaction of a Product. (Reason 1 for redundancies)
-    # Some fields/names are re-used for multiple things, such as filename for event label + item name dimension + product name + page title dimension (unusued) + ...
-    ga_payload = {
-        "v": 1,
-        "tid": ga_tid,                          # see ga_host_map
-        "t": "event",                           # Hit type. Could also be event, transaction, pageview, etc.
-        # Override IP address. Else will send detail about EC2 server which not too useful.
-        "uip": request.remote_addr,
-        "ua": request.user_agent,
-        "dl": request.url,
-        "dt" : filename,
-        # This is a ~ random ID/number. Used as fallback, since one is required
-        # if don't provided uid. While we still allow users to not be logged in,
-        # should at least be able to preserve/track their anon downloads..
-        "cid": "555",
-        "an": "4DN Data Portal EC2 Server",     # App name, unsure if used yet
-        "ec": "Serverside File Download",       # Event Category
-        "ea": "Range Query" if request.range else "File Download", # Event Action
-        "el": filename,                         # Event Label
-        "ev": file_size_downloaded,             # Event Value
-        # Product fields
-        "pa": "purchase",
-        "ti": str(uuid4()),                     # We need to send a unique transaction id along w. 'transactions' like purchases
-        "pr1id": file_at_id,                    # Product ID/SKU
-        "pr1nm": filename,                      # Product Name
-        "pr1br" : lab.get("display_title"),     # Product Branch
-        "pr1qt": 1,                             # Product Quantity
-        # Product Category from @type, e.g. "File/FileProcessed"
-        "pr1ca": "/".join([ ty for ty in reversed(context.jsonld_type()[:-1]) ]),
-        # Product "Variant" (supposed to be like black, gray, etc), we repurpose for filetype for reporting
-        "pr1va": properties.get("file_type", "other") # "other" MATCHES THAT IN `file_type_detaild` calc property, since file_type_detailed is used on frontend when performing "Select All" files.
-    }
+            ga_payload['cid'] = ga_cid
+
+        # TODO: WE SHOULD WRAP IN TRY/EXCEPT BLOCK RE: NETWORK?
```

Comment:
Yeah, I would probably do this just to be safe

Reply:
Only concern is then we might lose the data. Thinking maybe we should then have some fallback action that writes download action/info to like S3 bucket or something? Is server log(s) good enough?

Reply:
By lose data do you mean the GA event? I think this is very much a non issue but we should use an asynchronous retry mechanism for this piece that way the request does not block on posting a GA event.

Reply:
Yeah, like if GA server is down or something and can't connect to send the event.

Then we might have some untracked downloads occurring. Which, yeah, idk is arguably important or not. I'd lean on side of wanting to preserve all the datas but prly doesn't matter.

Currently I think might as well just fail the download if GA connection fails (prevent download) and ask end user to retry later. That way we at least have accuracy.. idk. Then once we have Kafka setup or something we can use that as a fallback and/or redundant store of downloads.